rclpy/test/test_node.py

```diff
@@ -66,6 +66,30 @@ def test_create_subscription(self):
         with self.assertRaisesRegex(ValueError, 'unknown substitution'):
             self.node.create_subscription(Primitives, 'foo/{bad_sub}', lambda msg: print(msg))
 
+    def raw_subscription_callback(self, msg):
+        print('Raw subscription callback: %s length %d' % (msg, len(msg)))
+        self.raw_subscription_msg = msg
+
+    def test_create_raw_subscription(self):
+        primitives_pub = self.node.create_publisher(Primitives, 'raw_subscription_test')
+        self.raw_subscription_msg = None  # None=No result yet
+        self.node.create_subscription(
+            Primitives,
+            'raw_subscription_test',
+            self.raw_subscription_callback,
+            raw=True
+        )
+        primitives_msg = Primitives()
+        cycle_count = 0
+        while rclpy.ok() and cycle_count < 5 and self.raw_subscription_msg is None:
+            primitives_pub.publish(primitives_msg)
+            cycle_count += 1
+            rclpy.spin_once(self.node, timeout_sec=1)
+        self.assertIsNotNone(self.raw_subscription_msg, 'raw subscribe timed out')
+        self.assertIs(type(self.raw_subscription_msg), bytes, 'raw subscribe did not return bytes')
+        # The length might be implementation dependant, but shouldn't be zero
+        self.assertNotEqual(len(self.raw_subscription_msg), 0, 'raw subscribe invalid length')
```

Comment:
Does this really test that the incoming message is complete and correct?

Reply:
It only partially tests the incoming message. 

That said, it seems unlikely that it's incorrect given that the rclpy code does not manipulate the array. If a non-zero array is passed up, it means that rcl_take_serialized_message returned bad data, and it should have its own tests. I understand the desire for completeness though, and if we do expose deserialization to rclpy we can make this test more robust.